### DISCUSSION TITLE
Is there a way to not have Quarto include escape character '\' before _ in template?

### DISCUSSION THREAD

**Discussion by DamonCharlesRoberts** [quarto-dev/quarto-cli#3098] [Q&A] [answered]
I created a template for myself and am having this issue where, when taking a file name that I provide in the yaml section of my Quarto document, it automatically places the \ escape character before an underscore symbol that I use for the file name in my LaTeX file. I'm not sure how clear this is coming across. So here's some examples:

In my quarto document, I have this:

`
bibliography: "current_projects/bibliography_file.bib"
`

I am then sending this through a .TeX template that takes the bibliography information and pastes it in:

`
\addbibresource{$for(bibliography)$$bibliography$$endfor$}%accessing the bilbiography file specified in the qmd
`

When I look at the intermediary TeX file it creates, somewhere along the line, Quarto automatically added the \ escape character before the underscore symbol in my file name. This, of course, makes it so that TinyTeX is unable to find that file. 

`
\addbibresource{current\_projects/bibliography\_file.bib}
`

Do I need to make an adjustment to my template tex file or is there some way to set this up in my Quarto document?

**Comment by mcanouil** [comment, marked as answer]
This is a LaTeX constraint about underscores (at least for most of LaTeX).
Quarto is only fixing the wrongly not escaped underscores.

Does your tex file compile if you remove the `\`?
If yes, then maybe Quarto is trying too hard to be smart, thus it might need to be fixed.

**Reply by DamonCharlesRoberts** [reply]
Once I get rid of the escape characters, everything compiles fine. 

I agree, I think Quarto just sees underscores and throws the \ in there.

Is that a feature request or is there a way to change this default behavior? It makes sense for Quarto to do that in most cases, but there should be a way for the user to overwrite that default behavior in situations like this.

**Reply by mcanouil** [reply]
It's a bug to me, not a feature request.
Quarto should not escape file path when used in LaTeX option (i.e., within brackets).

**Reply by dragonstyle** [reply]
I reproduce this with Pandoc itself - are the underscores actually causing issues with adding the bibresource? Similar (not identical, though) Pandoc issue:

https://github.com/jgm/pandoc/issues/7229

**Reply by DamonCharlesRoberts** [reply]
Hi @dragonstyle: I think the escape characters added to the underscores are the issue. I just tried compiling the TeX document itself with Tectonic and it has problems with the escape characters. But I have no problems compiling the document when I remove the escape characters and leave the underscores to the file path and name.

So this is a pandoc issue rather than a Quarto issue?

**Reply by dragonstyle** [reply]
OK, that is good to know- I do think it is a Pandoc issue as it is doing the template substitution under the hood. I created a simple test case with a markdown file and template to confirm that it is doing this substitution:

[Archive.zip](https://github.com/quarto-dev/quarto-cli/files/9901522/Archive.zip)

And then running 

```
pandoc test.md --from markdown --to latex --template template.tex
```

**Reply by DamonCharlesRoberts** [reply]
Gotcha. 

Okay, so should I basically just not use underscores then for my file names? Or to use mklatex as the pdf engine?

**Reply by dragonstyle** [reply]
Interesting - I'm actually not sure. Quarto can use TinyTex which provides some of the benefits of Tectonic and we default to using XeLaTeX as well - so you could simply try installing tinytex:

```quarto install tinytex```

and seeing whether rendered in this mode deals with the underscores properly (you could also experiment with using lualatex or pdflatex in this mode). If this worked, it would potentially point to an issue with Tectonic.

We did a lot of work on improving our TinyTex support in v1.2, so I could encourage you to use the Release Candidate if you want to give this route a try:

https://quarto.org/docs/download/

**Reply by DamonCharlesRoberts** [reply]
Looks like this happens with TinyTex too. I'll just use - instead of _ in my file names. Didn't know that about pandoc.

Thanks for the help @dragonstyle and @mcanouil !!

**Comment by Nenuial** [comment]
I put this here in case anybody finds this like I did.
I was looking for a way to have a path in an extension (they will inevitably have an underscore in the path because they are located in the `_extensions` folder) with the underscore that isn't escaped by pandoc.

I finally found that if the variable is defined under the `variables` key, then there will be no escaping. Example :

```yml
title: "Example extension"
author: ""
version: 1.1.0
contributes:
  format:
    pdf:
      variables:
        path-to-resource: "path/to/resource/"
```

You can then use the `$path-to-resource$` in the partials with no escaped underscore.
